build-and-push/action.yaml

```diff
@@ -0,0 +1,32 @@
+name: "build and push images"
+description: "build and push images do docker using docker compose"
+inputs:
+  location:
+    required: false
+    type: string
+  subdir:
+    required: false
+    type: string
+  tags:
+    required: false
+    type: string
+  type:
+    required: false
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Build and push ${{ inputs.type }}
+      env:
+        BUILD_LOCATION: ${{ inputs.location }}
```

Comment:
I am not sure about the name build_location, but can be persuaded. 

Is this taking place of the subdir?

Reply:
Oh I see /worker, so the image we are building? I am still unsure of the naming.